Autopilot/AttitudeManager/FixedWing.c

```diff
@@ -0,0 +1,248 @@
+    }
+    setProgramStatus(UNARMED);
+
+    while (getProgramStatus() == UNARMED){
+        StateMachine(STATEMACHINE_IDLE);
+    }
+    setProgramStatus(MAIN_EXECUTION);
+}
+
+void takeOff(){
+
+}
+void landing(){
+    
+}
+
+void inputMixing(int* channelIn, int* rollRate, int* pitchRate, int* throttle, int* yawRate){
+        if (getControlPermission(ROLL_CONTROL_SOURCE, ROLL_RC_SOURCE,ROLL_CONTROL_SOURCE_SHIFT)){
+            (*rollRate) = channelIn[ROLL_IN_CHANNEL - 1];
+        }
+        if (getControlPermission(THROTTLE_CONTROL_SOURCE, THROTTLE_RC_SOURCE,THROTTLE_CONTROL_SOURCE_SHIFT)) {
+            (*throttle) = (channelIn[THROTTLE_IN_CHANNEL - 1]);
+        }
+
+        #if TAIL_TYPE == STANDARD_TAIL
+        if (getControlPermission(PITCH_CONTROL_SOURCE, PITCH_RC_SOURCE,0)){
+            (*pitchRate) = channelIn[PITCH_IN_CHANNEL - 1];
+        }
+        (*yawRate) = channelIn[YAW_IN_CHANNEL - 1];
+        
+        #elif TAIL_TYPE == V_TAIL    //V-tail
+        // TODO
+
+        #elif TAIL_TYPE == INV_V_TAIL
+        if (getControlPermission(PITCH_CONTROL_SOURCE, PITCH_RC_SOURCE,0)){
+            (*pitchRate) = (channelIn[L_TAIL_IN_CHANNEL - 1] - channelIn[R_TAIL_IN_CHANNEL - 1]) / (2 * ELEVATOR_PROPORTION);
+        }
+        (*yawRate) = (channelIn[L_TAIL_IN_CHANNEL - 1] + channelIn[R_TAIL_IN_CHANNEL - 1] ) / (2 * RUDDER_PROPORTION);
+        
+        #endif
+}
+
+void outputMixing(int* channelOut, int* control_Roll, int* control_Pitch, int* control_Throttle, int* control_Yaw){
+    //code for different tail configurations
+    #if TAIL_TYPE == STANDARD_TAIL  //is a normal t-tail
+    channelsOut[PITCH_OUT_CHANNEL] = (*control_Pitch);
+    channelsOut[YAW_OUT_CHANNEL] = (*control_Yaw);
+
+    #elif TAIL_TYPE == V_TAIL    //V-tail
+    // TODO
+
+    #elif TAIL_TYPE == INV_V_TAIL   //Inverse V-Tail
+    channelOut[R_TAIL_OUT_CHANNEL - 1] =  (*control_Yaw) * RUDDER_PROPORTION + (*control_Pitch) * ELEVATOR_PROPORTION ; //Tail Output Right
+    channelOut[L_TAIL_OUT_CHANNEL - 1] =  (*control_Yaw) * RUDDER_PROPORTION - (*control_Pitch) * ELEVATOR_PROPORTION ; //Tail Output Left
+    
+    #endif
+    channelOut[ROLL_OUT_CHANNEL - 1] = (*control_Roll);
+    channelOut[THROTTLE_OUT_CHANNEL - 1] = (*control_Throttle);
+}
+
+void checkLimits(int* channelOut){
+    if (channelOut[ROLL_OUT_CHANNEL - 1] > MAX_ROLL_PWM) {
+        channelOut[ROLL_OUT_CHANNEL - 1] = MAX_ROLL_PWM;
+        // Limits the effects of the integrator, if the output signal is maxed out
+//        if (getIntegralSum(ROLL) * getGain(ROLL, GAIN_KI) > sp_RollRate - sp_ComputedRollRate) {
+//            setIntegralSum(ROLL, getIntegralSum(ROLL)/1.1);
+//        }
+    }
+    if (channelOut[ROLL_OUT_CHANNEL - 1] < MIN_ROLL_PWM) {
+        channelOut[ROLL_OUT_CHANNEL - 1] = MIN_ROLL_PWM;
+        // Limits the effects of the integrator, if the output signal is maxed out
+//        if (getIntegralSum(ROLL) * getGain(ROLL, GAIN_KI) < sp_RollAngle - sp_ComputedRollRate) {
```

Comment:
if the next PR is going to be ready soon, close this one and split it up into multiple PR's like i mentioned. Make sure that the PR's are logically seperated